### DISCUSSION TITLE
Option to disable coercing in `Model.eval`?

### DISCUSSION THREAD

**Discussion by kmnhan** [lmfit/lmfit-py#903] [General]
Hello,

Although array coercing has a much more consistent behavior with the 1.2.2 update, it happens to break some of my existing code that relies on `xarray`'s broadcasting.

For instance, when I have a `ModelResult` from a fit, I used to do something like `result.eval(x=arr)` where `arr` is a `xarray.DataArray`.
This would return a `xarray.DataArray` which retains the coordinates and attributes of `arr`. Now, the output is coerced into a numpy array.

Hence it would be great if the coercion for `eval` could be toggled off, for instance with a keyword argument, where `result.eval(x=arr, coerce_farray=False)` would have the old behavior. This would also mean more consistency with `Model.fit`.

**Comment by newville** [comment]
@kmnhan Can you just call your model function directly?   

I suppose `ModelResult.eval()` could have the option to suppress `coerce_farray`.  I'm not sure we want to do that. 

I am somewhat sympathetic, and we do try to allow "array-like" data on input, but `lmfit.ModelResult.eval` is a method of the `lmfit` classes, and we need to get the output of `eval()` to be a real ndarray.  But, we need solutions for #875 and #900.  

So, I guess I am a "maybe" on this...
